sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwaptionWithNamedConventionsExamples.cs

```diff
@@ -0,0 +1,136 @@
+            // For accurate pricing, one would want to upsert a quote per reset. 
+            var quoteRequest = TestDataUtilities.BuildQuoteRequest(
+                "BP00",
+                QuoteSeriesId.InstrumentIdTypeEnum.RIC,
+                0.05m,
+                "USD",
+                new DateTimeOffset(2020, 01, 01, 0, 0, 0, 0, TimeSpan.Zero));
+            var upsertResponse = _quotesApi.UpsertQuotes(scope, quoteRequest);
+            Assert.That(upsertResponse.Failed.Count, Is.EqualTo(0));
+            Assert.That(upsertResponse.Values.Count, Is.EqualTo(quoteRequest.Count));
+            
+            // For models requiring discount curves, we upsert them below. ConstantTimeValueOfMoney does not require any discount curves. 
+            var upsertComplexMarketDataRequest = new Dictionary<string, UpsertComplexMarketDataRequest>();
+            if (model != ModelSelection.ModelEnum.ConstantTimeValueOfMoney)
+            {
+                upsertComplexMarketDataRequest.Add("discount_curve", TestDataUtilities.BuildOisCurveRequest(TestDataUtilities.EffectiveAt, "USD"));
+                upsertComplexMarketDataRequest.Add("6M_rate_Curve", TestDataUtilities.Build6MRateCurveRequest(TestDataUtilities.EffectiveAt, "USD"));
+            }
+            if (model == ModelSelection.ModelEnum.BlackScholes || model == ModelSelection.ModelEnum.Bachelier)
+            {
+                var volatility = (model == ModelSelection.ModelEnum.BlackScholes) ? 0.2m : 10m;
+                upsertComplexMarketDataRequest.Add("VolSurface", TestDataUtilities.ConstantVolatilitySurfaceRequest(TestDataUtilities.EffectiveAt, instrument, model, volatility));
+            }
+            
+            if(upsertComplexMarketDataRequest.Any())
+            {
+                var upsertComplexMarketDataResponse = _complexMarketDataApi.UpsertComplexMarketData(scope, upsertComplexMarketDataRequest);
+                ValidateComplexMarketDataUpsert(upsertComplexMarketDataResponse, upsertComplexMarketDataRequest.Count);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void GetAndValidatePortfolioCashFlows(
+            LusidInstrument instrument,
+            string scope,
+            string portfolioCode,
+            string recipeCode,
+            string instrumentID)
+        {
+            var swaption = (InterestRateSwaption) instrument;
+            var cashflows = _transactionPortfoliosApi.GetPortfolioCashFlows(
+                scope: scope,
+                code: portfolioCode,
+                effectiveAt: TestDataUtilities.EffectiveAt,
+                windowStart: swaption.StartDate.AddDays(-3),
+                windowEnd: swaption.Swap.MaturityDate.AddDays(3),
+                asAt:null,
+                filter:null,
+                recipeIdScope: scope,
+                recipeIdCode: recipeCode).Values;
+            
+            Assert.That(cashflows.Count, Is.GreaterThanOrEqualTo(1));
```

Comment:
we should assert the cashflow on the date at a minimum

Reply:
I'm less sure. We've broken SDK before as a result of asserts like this. E.g. we recently made a change such that the cashflow of the bond's principal is adjusted to a business day. The same may occur for swaptions. Should SDK assert the (current) truth of the contents that has the possibility to change?